test-util/src/main/java/io/zeebe/test/util/bpmn/random/blocks/ProcessBuilder.java

```diff
@@ -56,12 +80,62 @@ public BpmnModelInstance buildProcess() {
     return processWorkInProgress.endEvent(endEventId).done();
   }
 
+  private void buildEventSubProcess(
+      final io.zeebe.model.bpmn.builder.ProcessBuilder processBuilder) {
+    processBuilder
+        .eventSubProcess(eventSubProcessId)
+        .startEvent("start_event_" + eventSubProcessId)
+        .interrupting(isEventSubProcessInterrupting)
+        .message(
+            b ->
+                // https://github.com/camunda-cloud/zeebe/issues/4099
+                // we might be not able to correlate a message when we have a message start event
+                // in the process
+                // as a workaround we use a constant string here
```

Comment:
I don't understand how the workaround works. Why is using a constant string helping? Also what if we had multiple event sub processes in the future?

Reply:
I think the problem is that when you have a message start event then variables are not correctly copied which will not trigger the event sub process then. If you use a static value you can trigger the event sub process.

@saig0 correct me if I'm wrong